presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -128,6 +137,49 @@ public Password()
         }
     }
 
+    private static class Roles
+            extends AbstractConnectionProperty<Map<String, ClientSelectedRole>>
+    {
+        public Roles()
+        {
+            super("roles", NOT_REQUIRED, ALLOWED, Roles::parseRoles);
+        }
+
+        // Roles consists of a list of catalog role pairs.
+        // E.g., `jdbc:presto://example.net:8080/?roles=catalog1:none;catalog2:all;catalog3:role` will set roles `none` in `catalog1`, `all` in `catalog2` and `role` in `catalog3`
+        public static Map<String, ClientSelectedRole> parseRoles(String roles)
+        {
+            return MAP_ENTRIES_SPLITTER.splitToList(roles).stream()
+                    .map(Roles::parseRole)
+                    .collect(toImmutableMap(entry -> entry.get(0), entry -> mapToClientSelectedRole(entry.get(1))));
+        }
+
+        private static ClientSelectedRole mapToClientSelectedRole(String role)
+        {
+            if (ALL.name().equalsIgnoreCase(role)) {
+                return new ClientSelectedRole(ALL, Optional.empty());
+            }
+            if (NONE.name().equalsIgnoreCase(role)) {
+                return new ClientSelectedRole(NONE, Optional.empty());
+            }
+            return new ClientSelectedRole(ClientSelectedRole.Type.ROLE, Optional.of(role));
+        }
+
+        public static List<String> parseRole(String catalogRoleEntry)
+        {
+            List<String> catalogRole = MAP_ENTRY_SPLITTER.splitToList(catalogRoleEntry);
+            checkArgument(catalogRole.size() == 2, "Malformed role: %s", catalogRoleEntry);
+            String catalog = catalogRole.get(0);
+            String role = urlDecode(catalogRole.get(1));
+            checkArgument(!catalog.isEmpty(), "Role catalog is empty");
+            checkArgument(!role.isEmpty(), "Role role is empty");
+
+            checkArgument(PRINTABLE_ASCII.matchesAllOf(catalog), "Role catalog contains spaces or is not printable ASCII: %s", catalog);
```

Comment:
If `catalog` contains a space, it may be trailing space. Quote offending value. 

Ideally we could do something about nonprintables (like https://github.com/prestosql/presto/blob/6a78c6023e008526d28344b177165bbbdeb8a5ed/presto-orc/src/main/java/io/prestosql/orc/metadata/statistics/StringStatistics.java#L45-L50), but I think it's overkill here.

Reply:
> it cannot contain space as it is out of range of printable characters

Yes, but you're reporting `catalog` value that does not pass the validation, so you assume it contains some "non printable" characters.
You need to quote exactly because of that.

BTW "PRINTABLE_ASCII" name is misleading. A space is definitely a printable character, but we want to exclude it from the allowed characters.
i don't know how to name this. @electrum do you?